### PR TITLE
array function, including the sorting function and cluster related function

### PR DESCRIPTION
**Pull request by zhangweikop**

### CODE REVIEW

File: source/core/TDCodecVia.cpp [hunk 1]
```diff
@@ -718,6 +718,27 @@ void TDViaParser::PreParseElements(Int32 rank, ArrayDimensionVector dimensionLen
...
+{
+   if (input->EatChar('{')) {
+       input->EatWhiteSpaces();
+       while (input->Length()>0 && !input->EatChar('}')) {
+           input->ReadToken(token);
+           input->EatWhiteSpaces();
+           if (!input->EatChar(*tsNameSuffix)) {
+               return TokenTraits_Unrecognized;
+           }
+           if (!ReadArrayItem(input, token)) { return TokenTraits_Unrecognized; }
```
Comment (PaulAustin) [inline source/core/TDCodecVia.cpp]: Use standard indention

File: source/core/vireo.preamble.js [hunk 2]
```diff
@@ -2,4 +2,6 @@ if (typeof(NationalInstruments) === "undefined") {
     NationalInstruments = {};
     }
 NationalInstruments.Vireo = (function() {
+    var Module = {};
+    Module.TOTAL_MEMORY = NationalInstruments.MemoryLimit || 524288000;
```
Comment (PaulAustin) [inline source/core/vireo.preamble.js]: What does this do?

File: source/core/TypeAndDataManager.cpp [hunk 3]
```diff
@@ -67,8 +67,9 @@ TypeManager::TypeManager(TypeManagerRef parentTm)
     _totalAQAllocated = 0;
     _totalAllocationFailures = 0;
     _maxAllocated = 0;
-    _allocationLimit = 16777216;  //16 meg for starters
-    
+    //_allocationLimit = 16777216;  //16 meg for starters
+    _allocationLimit = 524288000;  //500 meg
```
Comment (PaulAustin) [inline source/core/TypeAndDataManager.cpp]: This is a good time to make it configurable.  A fixed amount this large will make LV web pages a memory hog.

**Comment by cglzaguilar** [comment]
Wei is going to work on addressing these issues on separate pull requests. We agree on closing this pull request.

**Pull request closed without merging**